include/exadg/operators/operator_base.cpp

```diff
@@ -708,24 +696,12 @@ template<int dim, typename Number, int n_components>
 void
 OperatorBase<dim, Number, n_components>::init_system_matrix(SparseMatrix & system_matrix) const
 {
-  DoFHandler<dim> const & dof_handler = this->matrix_free->get_dof_handler(this->data.dof_index);
-
-  MPI_Comm comm;
-
-  // extract communicator
-  {
-    auto tria =
-      dynamic_cast<parallel::TriangulationBase<dim> const *>(&dof_handler.get_triangulation());
-
-    if(tria != NULL)
-      comm = tria->get_communicator();
-    else // not distributed triangulation
-      comm = MPI_COMM_SELF;
-  }
+  DoFHandler<dim> const &    dof_handler = this->matrix_free->get_dof_handler(this->data.dof_index);
+  Triangulation<dim> const & tria        = dof_handler.get_triangulation();
 
   TrilinosWrappers::SparsityPattern dsp(is_mg ? dof_handler.locally_owned_mg_dofs(this->level) :
                                                 dof_handler.locally_owned_dofs(),
-                                        comm);
+                                        tria.get_communicator());
```

Comment:
If `tria` is not used elsewhere in this function, I suggest to immediately use the `dof_handler.get_communicator()` function.